pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -117,9 +115,31 @@ void processArguments() throws Exception {}
      * Function level command
      */
     @Getter
-    abstract class FunctionCommand extends NamespaceCommand {
-        @Parameter(names = "--name", description = "The function's name", required = true)
+    abstract class FunctionCommand extends BaseCommand {
+        @Parameter(names = "--fqfn", description = "The Fully Qualified Function Name (FQFN) for the function")
+        protected String fqfn;
+
+        @Parameter(names = "--tenant", description = "The function's tenant")
+        protected String tenant;
+
+        @Parameter(names = "--namespace", description = "The function's namespace")
+        protected String namespace;
+
+        @Parameter(names = "--name", description = "The function's name")
         protected String functionName;
+
+        @Override
+        void processArguments() throws Exception {
```

Comment:
nit: it is good to call super.processArguments() in general. just in case people modified the logic in BaseCommand in future.

Reply:
I agree but in this case `FunctionCommand`s, `NamespaceCommand`s, and `FunctionConfigCommand`s really do require separate logic. I'm not sure I can see a way around that.

Reply:
Okay, never mind, I understand what you mean now. Fixed.